api/main_endpoints/routes/User.js

```diff
@@ -142,6 +145,31 @@ router.post('/edit', (req, res) => {
   } else if (!checkIfTokenValid(req)) {
     return res.sendStatus(UNAUTHORIZED);
   }
+
+  if(!req.body.email){
+    return res.sendStatus(BAD_REQUEST);
+  }
+
+  let decoded = decodeToken(req);
+  if(decoded.accessLevel === membershipState.MEMBER){
+    if(req.body.email && req.body.email != decoded.email){
+      return res
+        .status(UNAUTHORIZED)
+        .send('Unauthorized to edit another user');
+    }
+    if(req.body.accessLevel && req.body.accessLevel !== decoded.accessLevel){
+      return res
+        .status(UNAUTHORIZED)
+        .send('Unauthorized to change access level');
+    }
+  }
+
+  if(decoded.accessLevel === membershipState.OFFICER){
+    if(req.body.accessLevel && req.body.accessLevel == membershipState.ADMIN){
```

Comment:
I am not so sure about this line, nothing from the request body should be trusted server side. This logic look to first check the JWT token for the Officer access level (this is good) then check the request body for Admin level. This could allow an Officer to masquerade as a Admin.

Reply:
@aashrav clarified my misunderstanding here, resolving this.